app/src/main/java/com/example/myapplication/ThirdFragment2.java

```diff
@@ -45,13 +53,44 @@ public void onClick(View view) {
                         .navigate(R.id.action_thirdFragment2_to_SecondFragment);
             }
         });
+        ObjectAnimator animation = ObjectAnimator.ofFloat(binding.pop, "rotationX", new float[]{0f, 90f, 180f, 270f, 360f, 90f, 180f, 270f, 360f, 90f, 180f, 270f, 360f});
+        animation.setDuration(1000);
+        binding.pop.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                animation.start();
+                MainActivity.darkMode = !MainActivity.darkMode;
+                lightDark(v, MainActivity.darkMode);
+            }
+        });
 
+        lightDark(v, MainActivity.darkMode);
 
 
 
 
 
     }
-
+    public void lightDark (ViewGroup v, boolean mode){
+        int bgColor = (mode ? (Color.BLACK) : Color.WHITE);
+        int textColor = (mode ? (Color.WHITE) : (Color.BLACK));
+        for (int i = 0; i < v.getChildCount(); i ++){
+            View child = v.getChildAt(i);
+            if (!(child instanceof Button)) {
+                child.setBackgroundColor(bgColor);
+                if (child instanceof TextView) {
+                    TextView tx = (TextView) child;
+                    tx.setTextColor(textColor);
+                }
+            }
+            if (child instanceof Switch) {
+                Switch tx = (Switch) child;
+                tx.setTextColor(textColor);
+            }
+            if (child instanceof ViewGroup){
+                lightDark((ViewGroup)child, mode);
+            }
+        }
+    }
```

Comment:
This is exactly the same thing as in ThirdFragment.java. Since this method doesn't use any instances from the class, consider moving both to a helper class, and calling it instead.

For example, if you have a new .java class:

```
public class UIHelpers {
    public static void lightDark (ViewGroup v, boolean mode){
        int bgColor = (mode ? (Color.BLACK) : Color.WHITE);
        int textColor = (mode ? (Color.WHITE) : (Color.BLACK));
        for (int i = 0; i < v.getChildCount(); i ++){
            View child = v.getChildAt(i);
            if (!(child instanceof Button)) {
                child.setBackgroundColor(bgColor);
                if (child instanceof TextView) {
                    TextView tx = (TextView) child;
                    tx.setTextColor(textColor);
                }
            }
            if (child instanceof Switch) {
                Switch tx = (Switch) child;
                tx.setTextColor(textColor);
            }
            if (child instanceof ViewGroup){
                lightDark((ViewGroup)child, mode);
            }
        }
    }
}

```
You don't need to define this method in the ThirdFragment.java or ThirdFragment2.java. And when calling it, you can call `UIHelpers.lightDark(<arguments>)`

Reply:
fixed